pymoveit2/moveit2.py

```diff
@@ -1800,7 +1916,6 @@ def _send_goal_async_move_action(self):
             )
             return
 
-        self.__last_error_code = None
```

Comment:
I don't understand this change (and a similar change below). Is it causing some issues?

Reply:
Hmm, good point. I thought about it and don't know why I put that change in there. If I recall correctly, the issue I experienced was the following:

Say you are executing trajectories asynchronously. Your first trajectory succeeds. Then you execute another trajectory. Consider two cases: (1) the action server is unavailable; and (2) the action server accepts the goal and completes executing the trajectory very fast, before the asynchronous calling function is able to call `query_state()`. In both these cases, from the asynchronous calling function's perspective, it sent a trajectory to asynchronously execute and at the next timestamp, `query_state` returns `IDLE` with a error value of success. In other words, the calling function has no way of distinguishing the above two cases.

However, the above change is not the right fix to that issue. I shouldn't delete `self.__last_error_code = None`, but rather move it to before we check whether the action server is available.

Anyway, because of this confusion, I went ahead and removed those two lines of changes. I'll add them back into a future PR if I run into the issue again.

Reply:
Oh, I see. That makes sense. Thank you for the explanation!